pkg/controller/authentication/authentication_controller.go

```diff
@@ -259,6 +265,7 @@ func (r *ReconcileAuthentication) Reconcile(ctx context.Context, request reconci
 		install,
 		dexCfg,
 		r.clusterDomain,
+		disableDex,
```

Comment:
That would be nice, but the problem is we need to remove the dex components, and the easiest way to do that right now is to just create all the components with the dex config and delete them instead of creating them.

I'm not the _biggest_ fan of this solution, but other alternatives are either uglier with no benefit over this or are nicer but may require a little more refactoring of Dex.